### PR TITLE
Replaced pytz with zoneinfo

### PR DESCRIPTION
**Pull request by agronholm**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,12 +2,9 @@ os:
...
 python:
-  - "2.7"
   - "3.6"
   - "3.7"
   - "3.8"
   - "nightly"
 # command to run tests
-install:
-# Nothing needed
 script: python setup.py test
```
Comment (pganssle) [inline .travis.yml:10]: Not really @agronholm's problem but [`setup.py test` is deprecated](https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite) and will be removed in the relatively near future (though I'm disturbed to find that there is no deprecation warning in the Travis CI logs for some reason).

Comment (agronholm) [reply .travis.yml:10]: I'd be happy to overhaul the packaging and testing configuration but I figured it was beyond the scope of this PR.
For example, Travis only tests this on Linux even though there's Windows specific code.

File: setup.py [hunk 2]
```diff
@@ -35,10 +33,7 @@
       include_package_data=True,
       zip_safe=True,
       install_requires=[
-          'pytz',
-      ],
-      tests_require=[
-          'mock',
+          'backports.zoneinfo; python_version < "3.9"',
```
Comment (regebro) [inline setup.py:36]: Ah, backports.zoneinfo. Nice, I didn't know of that one. That makes this a lot better.

**Review by regebro — CHANGES_REQUESTED** [review]
This looks great. Could you add in CHANGES.txt that Python 2 support is dropped, and that we are using zoneinfo instead of pytz?

Also please add yourself to the contributor lists in README.txt

**Comment by regebro** [comment]
Weird that Travis-CI doesn't show up, but it passed anyway, so...

**Comment by agronholm** [comment]
I can make a follow-up PR that updates your testing and packaging configuration to use more modern practices, including running the test suite on multiple platforms using Github Actions. Travis still has pretty bad support for that.

**Review by regebro — APPROVED** [review]

**Comment by regebro** [comment]
Yeah, better testing would be great!

File: CHANGES.txt [hunk 3]
```diff
@@ -4,7 +4,8 @@ Changes
 2.2 (unreleased)
```
Comment (pganssle) [inline CHANGES.txt:4]: This should be version 3.0. This is a **major** breaking change and should break approximately everyone using this package.

Comment (pganssle) [reply CHANGES.txt:4]: That's good, though probably "switched timezone provider" is maybe a little under-stated here. I imagine that people won't realize that this completely changes the API of this package. There should probably be big bold text like, "If this change doesn't break you there's a good chance you were using this package wrong in the first place", and possibly a migration guide like the one in [`pytz-deprecation-shim`](https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html#migration-guide).

You could even use that one, or recommend that people who want a slower deprecation use `pytz-deprecation-shim`, either [`wrap_zone`](https://pytz-deprecation-shim.readthedocs.io/en/latest/api.html#pytz_deprecation_shim.wrap_zone) or [`timezone`](https://pytz-deprecation-shim.readthedocs.io/en/latest/api.html#pytz_deprecation_shim.timezone).

**Comment by agronholm** [comment]
I've created https://github.com/regebro/tzlocal/pull/94 for the testing configuration. I'm not sure why the Github Actions workflow was not triggered; I have the same trigger configuration in my other project and it does trigger for pull requests.

**Comment by agronholm** [comment]
I created https://github.com/regebro/tzlocal/pull/96 to fix the README that I forgot to update in this PR.

**Comment by regebro** [comment]
It has become obvious to me lately that this was a mistake. The question is how to fix it.
Please discuss:

https://github.com/regebro/tzlocal/issues/117

**Pull request merged**